src/Components/Bookshelf/Bookshelf.tsx

```diff
@@ -4,22 +4,26 @@ import Book from '../Book/Book'
 import '../Bookshelf/Bookshelf.css'
 
 type BookshelfProps = {
-  queryID: string
+  queryID?: string,
+  favoriteBooks?: Book[] | null,
+  addToFavorites?: (book: Book) => void
 }
 
-type BookshelfState = {
-    books: {
-    rank: number,
-    publisher: string,
-    description: string,
-    title: string,
-    author: string,
-    bookImage: string,
-    amazonProductUrl: string
-  }[] | null
+export interface Book {
+  rank: number,
+  publisher: string,
+  description: string,
+  title: string,
+  author: string,
+  bookImage: string,
+  amazonProductUrl: string
```

Comment:
Nice thinking with setting this up as an interface - now we can use it anywhere we need!